src/GraphVisualization.js

```diff
@@ -5,30 +5,79 @@ import { fetchTriples } from "./api";
 import { transformToGraphData } from "./graphData";
 import GraphLegend from "./GraphLegend";
 import GraphVR from "./GraphVR";
+import NodeDetailsSidebar from "./NodeDetailsSidebar";
 
 const GraphVisualization = () => {
   const [graphData, setGraphData] = useState({ nodes: [], links: [] });
   const [isInitialLoad, setIsInitialLoad] = useState(true);
   const [viewMode, setViewMode] = useState("2D");
+  const [selectedTriple, setSelectedTriple] = useState(null);
+  const [showCreators, setShowCreators] = useState(false);
   const fgRef = useRef();
 
-  // Fetch and transform graph data
+  // Charger les données
   useEffect(() => {
     const loadData = async () => {
       try {
         const triples = await fetchTriples();
-        const data = transformToGraphData(triples);
-        setGraphData(data);
+        let baseGraphData = transformToGraphData(triples);
+
+        // Ajouter les créateurs si le toggle est actif
+        if (showCreators) {
+          baseGraphData = enhanceGraphDataWithCreators(baseGraphData, triples);
+        }
+
+        setGraphData(baseGraphData);
       } catch (error) {
         console.error("Error loading graph data:", error);
       }
     };
+
     loadData();
-  }, []);
+  }, [showCreators]); // Recharger les données si le toggle `showCreators` change
+
+  // Fonction pour ajouter les créateurs au graphe
+  const enhanceGraphDataWithCreators = (graphData, triples) => {
+    const creatorNodes = [];
+    const creatorLinks = [];
+
+    triples.forEach((triple) => {
+      const entities = [triple.subject, triple.predicate, triple.object];
+
+      entities.forEach((entity) => {
+        if (entity.creatorId) {
+          // Ajouter un nœud pour le créateur
+          if (
+            !creatorNodes.find(
+              (node) => node.id === `creator-${entity.creatorId}`
+            )
+          ) {
+            creatorNodes.push({
+              id: `creator-${entity.creatorId}`,
+              label: `Creator ${entity.creatorId}`,
```

Comment:
as we already have the legend ( green -> creator ) I think we should remove "Creator" from the label